cvat/apps/engine/static/engine/js/shapes.js

```diff
@@ -209,6 +209,13 @@ class ShapeModel extends Listener {
         return counter;
     }
 
+    _clamp(value, min, max) {
+        if (window.frameClipper) {
+            return window.frameClipper.clamp(value, min, max);
+        }
+        return this.clipToFrame ? Math.clamp(value, min, max) : value;
```

Comment:
why can it be not present?

Reply:
I thought that shapes might be used not only in annotation UI so `frameClipper` would not be available. I've re-checked the code, they are only used in annotation UI so I simplified this function.